raft/doc.go

```diff
@@ -180,5 +180,89 @@ cannot be removed any more since the cluster cannot make progress.
 For this reason it is highly recommended to use three or more nodes in
 every cluster.
 
+MessageType
+
+Package raft sends and receives message in Protocol Buffer format (defined
+in raftpb package). Each state (follower, candidate, leader) implements its
+own Step method to advance with the given raftpb.Message. And each step is
+determined by its raftpb.MessageType, as follows:
+
```

Comment:
probably you want a common rule for the term handling. check https://github.com/coreos/etcd/blob/master/raft/raft.go#L487-L514

But actually I think we do not have to talk about how we are handing each message. We only need to talk about what they are. Or you are basically explaining raft algo in a msg centric way. If you want to make it correct, probably you need more than 100 lines of explanation.